### PR TITLE
[NVBUG-6280721][fix] Unwaive DeepSeek-V3.2 FP4 MTP perf-sanity test

### PR DESCRIPTION
**Pull request by xguannv**
…s fixed by #14891

These 5 perf_sanity test_e2e cases were waived for nvbugs/6280721 (post-merge 2765/2769 on 06/08-06/09), a flaky CUDA illegal memory access hit during the generation CUDA-graph capture of DeepSeek-V3.2-Exp FP4 with MTP draft.

Root cause: the DSA indexer DSL atom-split path was taken for MTP-Eagle draft (i>=1) calls where next_n collapses to 1, while the cached (dsl_expand_factor, dsl_atom) and the pre-built expanded buffers still describe the verify-time next_n. The mismatch drives the paged-MQA-logits kernel with inflated kv-lens / block-table, producing out-of-range top-k indices that convertReqIndexToGlobal (no upper-bound check) maps to an unmapped KV-pool row. PR #14891 (57413893af) guards the split on
`next_n == dsl_expand_factor * dsl_atom`, landing 06/10 -- after the waive was added, so the waive was never re-evaluated.

Local validation on clean main (no other change): B200 aggr configs v32_fp4_dep8_mtp1_8k1k and v32_fp4_tep8_mtp3_8k1k pass repeatedly (mtp3 4/4 in CI-faithful pytest runs). The remaining GB200 / disagg configs share the same config-agnostic root cause and fix; relying on CI to validate them.

## Summary by CodeRabbit

* **Tests**
  * Adjusted waived/skipped performance sanity test cases for certain DeepSeek v32 fp4 Blackwell scenarios.
  * Removed several no-longer-waived Blackwell variants and kept a smaller set of remaining waivers.
  * Updated waiver coverage for the upload-generation scenario by dropping one variant while preserving the others.

## Description

These waived perf sanity tests are related to flaky illegal memory access during CUDA-graph generation. The root cause in one line: when drafter is decoding, it shares the same config with verifier and thus adopting the wrong path where the verifier should go. This path assumes a longer sequence length than the true sequence length and led to accessing not existing kv cache ids. This problem is fixed by NVBUG-6241842's fix. 

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

**Comment by hyukn** [comment]
/bot run --post-merge --disable-fail-fast --stage-list "\*PerfSanity\*"

**Review by hyukn — APPROVED** [review]
